graphql-dgs-graphiql-autoconfigure/src/main/java/com/netflix/graphql/dgs/graphiql/autoconfiguration/GraphiQLConfigurer.kt

```diff
@@ -16,27 +16,63 @@
 
 package com.netflix.graphql.dgs.graphiql.autoconfiguration
 
+import com.netflix.graphql.dgs.graphiql.autoconfiguration.GraphiQLConfigurer.Constants.PATH_TO_GRAPHIQL_INDEX_HTML
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Configuration
-import org.springframework.web.servlet.config.annotation.EnableWebMvc
+import org.springframework.core.io.Resource
 import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
 import org.springframework.web.servlet.config.annotation.ViewControllerRegistry
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
 import org.springframework.web.servlet.resource.PathResourceResolver
+import org.springframework.web.servlet.resource.ResourceTransformer
+import org.springframework.web.servlet.resource.ResourceTransformerChain
+import org.springframework.web.servlet.resource.TransformedResource
+import java.io.BufferedReader
+import java.io.IOException
+import java.nio.charset.StandardCharsets.UTF_8
+import javax.servlet.http.HttpServletRequest
 
 @Configuration
-@EnableWebMvc
-open class GraphiQLConfigurer : WebMvcConfigurer {
+@Suppress("SpringJavaInjectionPointsAutowiringInspection")
+open class GraphiQLConfigurer(
+    @Value("\${dgs.graphql.graphiql.path:/graphiql}") private val graphiqlPath: String,
+    @Value("\${dgs.graphql.path:/graphql}") private val graphqlPath: String
+) : WebMvcConfigurer {
+
+    object Constants {
+        const val PATH_TO_GRAPHIQL_INDEX_HTML = "/graphiql/index.html"
+    }
+
+    override fun addViewControllers(registry: ViewControllerRegistry) {
+        registry.addViewController(graphiqlPath).setViewName("forward:$PATH_TO_GRAPHIQL_INDEX_HTML")
+        registry.addViewController("$graphiqlPath/").setViewName("forward:$PATH_TO_GRAPHIQL_INDEX_HTML")
```

Comment:
We somewhat rely on the paths not having a trailing slash - I added that effect in the metadata props but we are not really validating that anywhere. I plan on doing that validation in the config props.